src/cval.c

```diff
@@ -956,6 +956,9 @@ bool uint_inrange(largest_uint x, type t)
   if (tsize == sizeof(largest_uint) && type_unsigned(t))
     return TRUE;
 
+  if (type_Bool(t))
```

Comment:
Changing the *_inrange functions seems a) dangerous and b) likely the wrong place to achieve the desired effect.

I would remove the changes to these functions, and start by updating cval_cast to implement the cast-to-Bool rules - that will then automatically deal with, e.g., global initialisers (_Bool x = (void *) 0; or _Bool y = 2.3; e.g.). Some test cases (for compile-time-error vs no-compile-time-error) is then needed.


It does look like set_parameter_values in nesc-abstract.c should special-case bool, essentially replacing the special case in cval_inrange

Reply:
Done.